lib/web_ui/lib/src/engine/canvaskit/picture.dart

```diff
@@ -24,14 +30,38 @@ class CkPicture extends ManagedSkiaObject<SkPicture> implements ui.Picture {
   @override
   int get approximateBytesUsed => 0;
 
+  /// Whether the picture has been disposed of.
+  /// 
+  /// This is indended to be used in tests and assertions only.
+  bool get debugIsDisposed => _isDisposed;
+
+  /// This is set to true when [dispose] is called and is never reset back to
+  /// false.
+  /// 
+  /// This extra flag is necessary on top of [rawSkiaObject] because
+  /// [rawSkiaObject] being null does not indicate permanent deletion. See
+  /// similar flag [SkiaObjectBox.isDeletedPermanently].
+  bool _isDisposed = false;
+
   @override
   void dispose() {
+    if (_isDisposed) {
+      return;
```

Comment:
Isn't it a mistake to call dispose more than once? In the framework and on the VM side we generally assert not debug disposed.

Reply:
Good point. Changed to assert. I didn't see an assert in the AOT version, but I now empirically verified that it does throw an exception.